src/al-sap/al_service_registration_response.cpp

```diff
@@ -37,8 +38,10 @@ RegistrationResult AlServiceRegistrationResponse::getResult() const {
 
 std::vector<unsigned char> AlServiceRegistrationResponse::serializeRegistrationResponse() {
     std::vector<unsigned char> data;
-    
+
+    uint32_t packet_size = alMacAddressLocal.size() + 4 + 1;
```

Comment:
Magic numbers. My suggestion is merely a guess
```suggestion
    //                   = MAC length + size (4 bytes) + ?
    uint32_t packet_size = alMacAddressLocal.size() + sizeof(uint32_t) + sizeof(uint8_t);
```

Reply:
Addressed in the latest commit